### ISSUE TITLE
Histogram give erroneous result when run with extreme values in certain verbose values

### ISSUE THREAD

**Issue by eyalshimony** [GenericMappingTools/pygmt#1873]
I ran the following code:

xdiffs = [1074.8299202938215, 831.2858709770371, -54.62688683188753, 1296.071473310527, -20.53646894247504, 961.3857325421413, 395.1748918511439, 1429.4308986734832, 1217.3091173375142, 321.4067696753773, 105.46202416112646, 628.3520597061142, 67.83012599434005, 2266.9607700653723, 318.9331428001169, 371.4866164161358, 1400.5830340071698, 1395.3601524578407, -264.8147500117775, 560.6100958238239, 654.540597606916, -374.19939977891045, 102.39119975984795, 134.80147007008782, 884.2441466324381, -512.7298086992232, 261.3615814675577, 1225.7542172785616, 1622.0260769819724, 1260.0168619235046, 570.2609860597877, -397.6613760092878, 585.1282900830265, 2305.1443612164585, 1277.0824010685901, 263.2157081980258, 143.7260322398506, 116.65798494865885, -61.986646171309985, 94.35950381070143, 1794.8784583822708, 3522.6844897696283, 1094.3507381796371, 1415.6260909099365, -269.1036189328297, 51.522829379246105, 509.85441875836113, 567.7022887142957, 1169.126781546278, 48.98435432021506, 208.814438431873, 37.283830633736216, 8349.936332627141, 1009.9322088282206, 977.0698352382169, -240.9840025677695, 3378.604415353271, 42.68184827757068, -239.86246054712683, 864.4517739441362, 687.1352998627117, 556.5701368913869, -2344.9511524513946, 2086.75789618626, 2790.932233081083, 380.03524937218754, -2844.4500100392615, 1740.5896450100117, 889.39566773345, 1414.704840157705, -764.1091259552632, 825.191432279651, 2026.780120779411, 1685.9205319387838, -205.3036262911628, 1235.6085233052145, 1067.6139106209157, 1914.289638960443, 222.7017345284694, 1388.0504650576622, -600.6721038771211, 782.2806088114157, 846.7438859072863, 1674.6683126216521, -251.42987769207684, 1737.4857154254569, 920.1648562750197, 1415.654681094049, -309.92270781687694, 2223.5873799822875, 203.785953516257, 1289.671440719685, 793.3116924362257, -99.40296596515691, 1143.8520009944914, 1548.7048947695293, 1374.3960690958193, 373.78323516470846, -280.4147770701675, -361.78964831260964, 1034.5904550057603, -285.25429484498454, 2715.5492156649125, -96.14123875449877, 505.13876642187824, -1338.5473394987057, 65.44438167923363, 2959.865661800257, 125.6483726107399, 465.7149015624309, 3054.5074665169814, 276.2280071480782, 368.5888354406925, -389.7784073871444, 1591.4889496422838, 928.6415764052654, 792.3061098741018, -1523.9245277641458, 1014.8487145220861, 1360.6046203958103, 1663.9565467325738, -2126.6101516738418, 457.6914693908184, 734.3135498538031, -627.5367938916897, -79.50742841226747, 913.877365810622, -7241.458180868125, 216.45853887323756, -225.39877917914419, 187.68946576298913, 768.4510455300915, 2252.6869357005926, 725.8757694280939, -107.91023248678539, -1494.9329486220959, 2138.134616233059, 703.9465034956811, 1868.9548147112364, 416.87042135186493, 14459.13858444849, -104.64121229556622, 346.3747514920542, -132.42247643467272, 1978.322135917144, 123.4017646809807, -350.17842194624245, 585.2936880740453, 1214.5486482156557, 1699.444927606266, 634.1706474506645, 450.745337336557, 1392.0966321179294, 172.41508516052272, 1694.2192175981472, 490.59202523500426, 3280.098852172494, 831.2318677940057, 2497.6558223444736, 1257.6167378905811, 1959.8358380525024, 1057.3235051285592, 2537.828990992857, 827.696502478444, 945.8281042729504, -103.58591604535468, 715.7060685278266, 482.8399174187798, 568.2165091569768, -231.20547139504924, 1962.0570825885516, 1947.9156167460023, 2330.0284788424033, -452.3598039323697, 110.85907459055306, 1866.8028410342522, 606.6013040469261, 1572.2382799184998, -34.0320024737739, 18.93945101572899, 746.1261592697701, 918.5752789258258, 281.96332439174876, 1442.1241249716259, 386.6531754307798, 1827.815216950432, 754.1041557039716, -355.7393620974617, 969.749157001148, 1071.2471767785028, 1247.1626197996666, -11135.544669253111, 1812.3758272883715, 788.708030880196, -189.3510818867362, 1415.3017558662686, 181.8167145922198, 997.7844123126124, 324.0910986244562, -112.04958823206834, 95.42949640395818, 1253.8784322560532, 122.04925882990938, 752.3934782948345, 1046.551866867987, 1558.5735387297464, 2040.3052794309915, 712.2510732205119, -424.1431641824893, 87.75360091222683, 64.1622100333334, 1506.26450356422, 936.8970407794695, -225.87299600237748, 266.5185027755215, 1229.6623108519707, 2946.7314064149396, 570.9396867480827, 652.5001936289482, 219.6902991780662, 731.0174074423267, 797.2955506356666, -7165.203889584227, 1279.7580423504114, 632.8448451334843, 1100.6016587209888, 387.6417454559705, -5559.302793647337, 425.0010947200353, 999.9686572718201, 195.18599144677864, 999.0847336847219, 209.08241103834007, 1499.282315121498, 1814.2271667482564, 465.12400393298594, 1132.5552866558428, 613.6991897816188, 1094.428060147795, 2333.513462945004, 1581.0766488329973, 462.29942722507985, 817.1516242270591, 679.6556303598336, 641.5946589562809, 978.750222577597, 491.1038386663422, 796.8978282518801, 337.897966976685, 1093.6382728090393, 1025.3555918548955, 1245.8112093394157, 478.5636854590848, 691.9410134049831, 991.5239536367008, 1053.9136437688721, 762.5700777240563, 1771.3844914889196, 608.5016374623519, 880.0470970771858, -34.88717683649156, 298.7940502425772, -423.44739574572304, 524.4641876333044, 406.17551041865954, -800.5088671318372, 1381.845823425334, 3351.411926840141, 304.92292128049303, 480.2723048646585, 768.9319656862062, -308.79606238531414, 2182.4105561721954, 742.218788397091, 7270.0334851969965, -387.8567795025883, 1764.664728022879, 1093.7827999592992, 1179.1403678563074, 1963.064278539503, 397.700331635715, 483.9670632139896, 272.81123458675575, 248.9491789775784, 517.4890077930759, 1735.0264054206782, 470.33959560096264, 1900.007300931844, 3170.1096927622566, 607.1650719346944, 649.1383944115369, 2173.4748333300813, 645.9448107436183, 132.26746295019984, -2116.3652937777806, 267.7961256947601, 1091.6661670545582, 537.561562121613, 1518.028641193523, 692.5000184341334, 1157.6157334347954, 908.8893531512003, -2187.6637714271783, 1032.7717209461844, 1128.36532698886, 265.79850516008446, 445.2690238954965, 1289.0895767380134, 1353.2433614248876, 1323.878045941412, 8882.60949984123, 1708.6028799604974, 1185.9528560317121, 825.6623176994617, 1260.9982367188204, 435.93655147985555, 1521.6826971066766, 683.5599801895441, 752.2995064376737, 1267.8382640144555, -1010.4887944379007, 2207.930619232473, 971.6867978996015, 1042.0491197673837, 2078.7280052446295, 1972.0699746280443, 1408.010666918708, 1834.8797677009716, 906.4260382628418, 378.86629223969067, 1384.0856976383366, 950.5051392405294, 216.36693450569874, 700.7754164697835, 1292.5374383773888, 1871.1855088842567, 834.8652816427057, 553.932033997844, 108.6649844183703, 796.6101298095309]
fig = pygmt.Figure()
fig.histogram(
    data=xdiffs,
    frame=['WSne', 'x+l"Distance (m)"', 'y+l"Counts"'],
    series=[-5000, 5000, 200],
    extreme="b",
    fill="red3",
    pen="1p",
    histtype=0
)
fig.show()

and got the following histogram:
![image](https://user-images.githubusercontent.com/80954047/162182434-373140cf-9fc2-4c7c-b9e6-664db9965703.png)
which is erroneous as there are only 4 values above 5000. This issue also happens when each of the verbose values q, e, w, t is set explicitly.
However, when the same command is run with verbose values of i, c, d the produced histogram is right.

**System information**

PyGMT information:
  version: v0.6.0
System information:
  python: 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
  executable: C:\Users\eyals\AppData\Local\Programs\Python\Python310\python.exe
  machine: Windows-10-10.0.19044-SP0
Dependency information:
  numpy: 1.21.2
  pandas: 1.3.3
  xarray: 0.19.0
  netCDF4: 1.5.7
  packaging: 21.0
  ghostscript: 9.54.0
  gmt: 6.3.0
GMT library information:
  binary dir: C:/Users/eyals/AppData/Local/Programs/Python/Python310
  cores: 4
  grid layout: rows
  library path: c:/programs/gmt6/bin/gmt_w64.dll
  padding: 2
  plugin dir: c:/programs/gmt6/bin/gmt_plugins
  share dir: c:/programs/gmt6/share
  version: 6.3.0

**Comment by seisman** [comment]
I can reproduce the issue using GMT CLI, so it's an upstream GMT bug. FYI, I'm using GMT `6.4.0_a8bbbcf_2022.04.01`: 
```
# incorrect 
$ gmt histogram input.txt -T-5000/5000/200 -BWSen -Bxa1000f -W1p -Gred3 -Lb -pdf map

# correct
$ gmt histogram input.txt -T-5000/5000/200 -BWSen -Bxa1000f -W1p -Gred3 -Lb -pdf map -Vd
```
The input data file is attached: [input.txt](https://github.com/GenericMappingTools/pygmt/files/8442357/input.txt)

@PaulWessel @meghanrjones

**Comment by PaulWessel** [comment]
Gotta be the check on Line 1044.  So we will need to do that work regardless of -V but of course only report the printings given the levels.  SHould be an easy fix, with luck before the community meeting - will look now.

**Comment by seisman** [comment]
Closed by #6532
